internal/uvm/network.go

```diff
@@ -24,8 +26,228 @@ var (
 	// ErrNetNSNotFound is an error indicating the guest UVM does not have a
 	// network namespace by this id.
 	ErrNetNSNotFound = errors.New("network namespace not found")
+	// ErrNICNotFound is an error indicating that the guest UVM does not have a NIC
+	// by this id.
+	ErrNICNotFound = errors.New("NIC not found in network namespace")
+)
+
+// Network namespace setup is a bit different for templates and clones.
+// For templates and clones we use a special network namespace ID.
+// Details about this can be found in the Networking section of the late-clone wiki page.
+//
+// In this function we take the namespace ID of the namespace that was created for this
+// UVM. We hot add the namespace (with the default ID if this is a template). We get the
+// endpoints associated with this namespace and then hot add those endpoints (by changing
+// their namespace IDs by the deafult IDs if it is a template).
+func SetupNetworkNamespace(ctx context.Context, vm *UtilityVM, nsid string) error {
```

Comment:
Now that this is moved into the uvm package, I think we should un-export it and make it a function on the UVM struct.

Reply:
Guess we don't really need to do that in this PR though

Reply:
I take it back, this would be a small change so it would be nice to do in this PR :)